lib/puppet/indirector/facts/facter.rb

```diff
@@ -71,18 +76,33 @@ def self.setup_external_search_paths(request)
     request.environment.modules.each do |m|
       if m.has_external_facts?
         dir = m.plugin_fact_directory
-        Puppet.debug { "Loading external facts from #{dir}" }
+        Puppet.debug "Loading external facts from #{dir}"
```

Comment:
This is a subtle thing, but the `debug` method accepts either a string or a block. If you pass a string, then puppet is forced to interpolate the string, only to be discarded when not running at the debug level. The block form doesn't have this issue, since the block is only yielded to when running at the debug level. I've been trying to convert puppet over to using the block form more consistently, but there are still lots of places that use the string form.